# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLLegendElement.h
# Commit: 2d16e9e90401
# Full Hash: 2d16e9e904010ba3389aad9729bac4f3744a588d
# Author: Mats Palmgren <mats@mozilla.com>
# Date: 2021-01-30 05:43:48
# Regressor Bug: 1683748
# File Overlap Count: 3
# Description:
#   Bug 1683748 - Support Grid/Flex/Table/Column layout for the rendered legend of a fieldset.  r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101555
# ==============================================================================

diff -r 5b99d9a7ad8d -r 2d16e9e90401 dom/html/HTMLLegendElement.h
--- a/dom/html/HTMLLegendElement.h	Fri Jan 29 11:30:44 2021 +0000
+++ b/dom/html/HTMLLegendElement.h	Fri Jan 29 15:45:42 2021 +0000
@@ -57,6 +57,16 @@
     InlineStart,
     InlineEnd,
   };
+
+  /**
+   * Return the align value to use for the given fieldset writing-mode.
+   * (This method resolves Left/Right to the appropriate InlineStart/InlineEnd).
+   * @param aCBWM the fieldset writing-mode
+   * @note we only parse left/right/center, so this method returns Center,
+   * InlineStart or InlineEnd.
+   */
+  LegendAlignValue LogicalAlign(mozilla::WritingMode aCBWM) const;
+
   /**
    * WebIDL Interface
    */